FormalConjectures/ForMathlib/Combinatorics/Basic.lean

```diff
@@ -20,19 +20,13 @@ import Mathlib.Order.Defs.PartialOrder
 
 open Function Set
 
-/-- A Sidon sequence (also called a Sidon set) is a sequence of natural numbers $a_0, a_1, ...$
-such that all pairwise sums $a_i + a_j$ are distinct for $i ≤ j$.
+variable {α γ: Type*} [AddCommMonoid α] [Membership α γ]
```

Comment:
I would suggest specialising to gamma = Set alpha, the extra generality here is essentially spurious

Reply:
I had `A : Finset ℕ` in mind, because there seem to be a few problems around that. Not having the extra generality would force us to write
`IsSidon A.toSet` instead of `IsSidon A`, (and `IsSidon ↑A` also doesn't work) which seemed to justify the extra generality. What do you think?

Reply:
I appreciate this point, but this is the general pattern in mathlib: we have `Set.Pairwise` and when we want to talk about a relation holding pairwise on a finset, we use `.toSet` for the conversion.